src/probnum/quad/_integration_measures.py

```diff
@@ -0,0 +1,267 @@
+                2a. If both ``domain[0]`` and ``domain[1]`` are scalars, they are
+                    expanded to constant vectors of length ``dim``.
+                2b. If only one of `domain[0]`` and ``domain[1]`` is a scalar and the
+                    length of the other equals ``dim``, the scalar one is expanded to a
+                    constant vector of length ``dim``.
+                2c. Otherwise, if neither of ``domain[0]`` and ``domain[1]`` is a
+                    scalar, error is raised if either of them has length which does not
+                    equal ``dim``.
+        """
+        domain_a_dim = np.size(domain[0])
+        domain_b_dim = np.size(domain[1])
+
+        # Check that given dimensions match and are positive
+        dim_mismatch = False
+        if dim is None:
+            if domain_a_dim == domain_b_dim:
+                dim = domain_a_dim
+            elif domain_a_dim == 1 or domain_b_dim == 1:
+                dim = np.max([domain_a_dim, domain_b_dim])
+            else:
+                dim_mismatch = True
+        else:
+            if (domain_a_dim > 1 or domain_b_dim > 1) and dim != np.max(
+                [domain_a_dim, domain_b_dim]
+            ):
+                dim_mismatch = True
+
+        if dim_mismatch:
+            raise ValueError(
+                "Domain limits must have the same length or at least "
+                "one of them has to be one-dimensional."
+            )
+        if dim < 1:
+            raise ValueError(f"Domain dimension dim = {dim} must be positive.")
+
+        # Use same domain limit in all dimensions if only one limit is given
+        if domain_a_dim == 1:
+            domain_a = np.full((dim,), domain[0])
+        else:
+            domain_a = domain[0]
+        if domain_b_dim == 1:
+            domain_b = np.full((dim,), domain[1])
+        else:
+            domain_b = domain[1]
+
+        # Check that the domain is non-empty
+        if not np.all(domain_a < domain_b):
+            raise ValueError("Domain must be non-empty.")
+
+        self.dim = dim
+        self.domain = (domain_a, domain_b)
+
+
+class LebesgueMeasure(IntegrationMeasure):
+    """Lebesgue measure on a hyper-rectangle.
+
+    Parameters
+    ----------
+    dim :
+        Dimension of the integration domain
+    domain :
+        Tuple which contains two arrays which define the start and end points,
+        respectively, of the rectangular integration domain.
+    normalized :
+         Boolean which controls whether or not the measure is normalized (i.e.,
+         integral over the domain is one)
+    random_state :
+        Random state of the random variable corresponding to the integration measure.
+    """
+
+    def __init__(
+        self,
+        domain: Tuple[Union[np.ndarray, FloatArgType], Union[np.ndarray, FloatArgType]],
+        dim: Optional[IntArgType] = None,
+        normalized: Optional[bool] = False,
+        random_state: Optional[RandomStateArgType] = None,
```

Comment:
The random state is only ever used in the `sample` method. I recommend making it an argument of `sample` such that `LebesgueMeasure` can be made immutable. This change will be made in `RandomVariable` in the future as well as discussed in one of the recent software meetings.

Reply:
I agree, but with the current setup of `RandomVariable` I don't know how to pass `random_state` other than when initializing `RandomVariable`. `RandomVariable.sample` does not accept a `random_state`. I think this needs action in `RandomVariable` first.